src/main/java/org/apache/sling/xss/impl/CustomPolicy.java

```diff
@@ -0,0 +1,261 @@
+                        for (String allowedValue : allowedValuesFromAttribute) {
+                            policyBuilder.allowAttributes(attribute.getName()).matching(true, allowedValue)
+                                    .onElements(tag.getValue().getName());
+                        }
+
+                        List<Pattern> regexsFromAttribute = attribute.getPatternList();
+                        if (!regexsFromAttribute.isEmpty()) {
+                            policyBuilder.allowAttributes(attribute.getName())
+                                    .matching(matchesToPatterns(regexsFromAttribute))
+                                    .onElements(tag.getValue().getName());
+                        } else {
+                            policyBuilder.allowAttributes(attribute.getName()).onElements(tag.getValue().getName());
+                        }
+                    }
+
+                    if (!styleSeen) {
+                        policyBuilder.allowAttributes(CssValidator.STYLE_ATTRIBUTE_NAME)
+                                .matching(cssValidator.newCssAttributePolicy()).onElements(tag.getValue().getName());
+                    }
+                    break;
+
+                default:
+                    throw new RuntimeException("No tag action found.");
+            }
+        }
+
+        // disallow style tag on specific elements
+        policyBuilder.disallowAttributes(CssValidator.STYLE_ATTRIBUTE_NAME)
+                .onElements(cssValidator.getDisallowedTagNames().toArray(new String[0]));
+
+        // ---------- dynamic attributes ------------
+        Map<String, Attribute> dynamicAttributes = new HashMap<>();
+
+        // checks if the dynamic attributes are allowed
+        if (policy.getDirectives().get(ALLOW_DYNAMIC_ATTRIBUTES).equals("true")) {
+            dynamicAttributes.putAll(policy.getDynamicAttributes());
+            for (Attribute attribute : dynamicAttributes.values()) {
+                if (attribute.getOnInvalid().equals(REMOVE_TAG_ONINVALID_ACTION)) {
+                    onInvalidRemoveTagList.add(attribute.getName());
+                }
+
+                List<Pattern> regexsFromAttribute = attribute.getPatternList();
+                for (Pattern regex : regexsFromAttribute) {
+                    dynamicAttributesPolicyMap.put(attribute.getName(), newDynamicAttributePolicy(regex));
+                }
+
+                List<String> allowedValuesFromAttribute = attribute.getLiterals();
+                if (!allowedValuesFromAttribute.isEmpty()) {
+                    dynamicAttributesPolicyMap.put(attribute.getName(),
+                            newDynamicAttributePolicy(true, allowedValuesFromAttribute.toArray(new String[0])));
+                }
+
+            }
+        }
+
+        policyFactory = policyBuilder.allowTextIn(CssValidator.STYLE_TAG_NAME).toFactory();
+
+    }
+
+    public PolicyFactory getCustomPolicyFactory() {
```

Comment:
I would rename this to 'getHtmlCleanerPolicyFactory', because we use the word "policy" a lot, and this clarifies why a "Policy" class has a "getPolicyFactory" method.

Reply:
Resolved: 99bb91ab6e2529fd6ef25858f874a26cb159e87e